docs/rule/no-classic-invocation-component.md

```diff
@@ -0,0 +1,64 @@
+## Use angle bracket invocation for components
+
+### Rule name: `no-classic-invocation-component`
+
+There are two ways to invoke a component in a template: classic invocation syntax (`{{my-component}}`), and angle bracket invocation syntax (`<MyComponent />`). The difference between them is syntactical. You should favour angle bracket invocation syntax as it improves readability of templates, i.e. disambiguates components from helpers, and is also the future direction Ember is going with Glimmer components.
+
+#### Bad
+
+```hbs
+{{bad-code}}
+{{#bad-code}}{{/bad-code}}
+{{nested/bad-code}}
+{{#nested/bad-code}}{{/nested/bad-code}}
+```
+
+#### Good
+
+```hbs
+<GoodCode />
+<GoodCode></GoodCode>
+<Nested::GoodCode />
+<Nested::GoodCode></Nested::GoodCode>
+
+{{! whitelisted helpers}}
+{{some-valid-helper param}}
+{{some/some-valid-helper param}}
+
+{{! in-built helpers}}
+{{if someProperty "yay"}}
+{{#each items as |item|}}
+  {{item}}
+{{/each}}
+```
+
+### Whitelisting helpers
```

Comment:
Can you tweak this to make it clear that it only impacts helpers directly used in mustaches (as the `PathExpression`, but don't say that since it's way too "in the weeds")? 

I think _for the most part_ folks don't use many helpers like that...